lab-spencer/lib/storage.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+let uuid = require('uuid');
+let fs = require('fs-extra');
+
+let storage = module.exports = {};
+
+storage.setItem = data => {
+  data.id = uuid.v1();
+  return fs.writeJson(`${__dirname}/../data/${data.id}`, data)
+    .then(() => Promise.resolve(data));
```

Comment:
Here you do need to do the manual return since `fs.writeJson` does not resolve the data it writes. You don't need to wrap it in `Promise.resolve` however, since the promise will automatically return whatever you return from the function.